### PR TITLE
Navigation item additional methods

### PR DESCRIPTION
**Pull request by ousid**
Based on the discussion in https://github.com/filamentphp/filament/discussions/6566, this PR add the ability to show/hide the custom navigation items, by setting `visible` or `hidden` methods to the custom item.

### Example
```php
use Filament\Facades\Filament;
use Filament\Navigation\NavigationItem;

Filament::serving(function () {
    Filament::registerNavigationItems([
        NavigationItem::make('Analytics')
            ->visible(fn (): bool => false)
            // or
            ->hidden(fn (): bool => true),
    ]);
});
```

### CODE REVIEW

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 1]
```diff
@@ -4,11 +4,18 @@
     'badge' => null,
     'badgeColor' => null,
     'icon',
+    'visible' => null,
+    'hidden' => null,
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: I don't think any views need to be changed to accommodate this. If a navigation item is hidden, then it should not be in the array of items in the first place

File: packages/admin/src/Navigation/NavigationItem.php [hunk 2]
```diff
@@ -26,6 +26,10 @@ class NavigationItem
 
     protected string | Closure | null $url = null;
 
+    protected ?bool $isHidden = null;
+
+    protected ?bool $isVisible = null;
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
    protected bool $isHidden = null;

    protected bool $isVisible = true;
```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 3]
```diff
@@ -60,6 +64,24 @@ public function icon(string $icon): static
...
+    }
+
+    public function hidden(bool | Closure $condition = null): static
+    {
+        $this->isHidden = is_callable($condition)
+                                ? $condition()
+                                : $condition;
+
+        return $this;
+    }
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
    public function visible(bool | Closure $condition = true): static
    {
        $this->isVisible = value($condition);

        return $this;
    }

    public function hidden(bool | Closure $condition = true): static
    {
        $this->isHidden = value($condition);

        return $this;
    }
```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 4]
```diff
@@ -123,6 +145,16 @@ public function getIcon(): string
         return $this->icon;
     }
 
+    public function getVisible(): ?bool
+    {
+        return $this->isVisible;
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
        return ! $this->isHidden();
```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 5]
```diff
@@ -123,6 +145,16 @@ public function getIcon(): string
...
+    }
+
+    public function getHidden(): ?bool
+    {
+        return $this->isHidden;
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
        if ($this->evaluate($this->isHidden)) {
            return true;
        }

        return ! $this->evaluate($this->isVisible);
```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 6]
```diff
@@ -123,6 +145,16 @@ public function getIcon(): string
         return $this->icon;
     }
 
+    public function getVisible(): ?bool
+    {
+        return $this->isVisible;
+    }
+
+    public function getHidden(): ?bool
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
    public function getHidden(): bool
```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 7]
```diff
@@ -123,6 +145,16 @@ public function getIcon(): string
         return $this->icon;
     }
 
+    public function getVisible(): ?bool
```
Comment (danharrin) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion
    public function getVisible(): bool
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php (hunk 1 above)
Comment (danharrin) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: So these changes should be removed and replaced with changes in the code that returns the navigation structure. Test should also be updated

**Review by danharrin — CHANGES_REQUESTED** [review]

**Comment by danharrin** [comment]
Please mark this as `Ready for review` when the refactor is complete as requested :)

**Comment by ousid** [comment]
Hi, Dan. 

Sure thing, I'll update the PR As soon as I can :)

File: packages/admin/resources/views/components/layouts/app/sidebar/group.blade.php [hunk 8]
```diff
@@ -64,17 +64,19 @@ class="ml-1 h-3 w-3 flex-shrink-0"
...
-                    :icon="$item->getIcon()"
-                    :active-icon="$item->getActiveIcon()"
-                    :url="$item->getUrl()"
-                    :badge="$item->getBadge()"
-                    :badgeColor="$item->getBadgeColor()"
-                    :shouldOpenUrlInNewTab="$item->shouldOpenUrlInNewTab()"
-                >
-                    {{ $item->getLabel() }}
-                </x-filament::layouts.app.sidebar.item>
+                @if ($item->getVisible() || ! $item->getHidden())
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/group.blade.php]:
```suggestion
                @if ($item->getVisible())
```

**Review by danharrin — CHANGES_REQUESTED** [review]

**Comment by danharrin** [comment]
Thanks

**Pull request merged**